modules/docs/arrow-docs/docs/docs/typeclasses/traverse/README.md

```diff
@@ -9,8 +9,259 @@ permalink: /docs/typeclasses/traverse/
+
+```kotlin:ank:silent
+object ValidEffects : OurSideEffectsFunctions {
+
+    override fun validateLogin(cred: Credential): Either<SecurityError, Unit> {
+        return Either.right(Unit)
+    }
+
+    override fun userInfo(user: User): DeferredK<Profile> {
+        return DeferredK.async { Profile(id = user.id) }
+    }
+
+    fun savingProfiles(): DeferredK<Unit>  = GlobalScope.async(Dispatchers.Default, CoroutineStart.LAZY){Unit}.k()
+
+}
+```
+
+We just defined results for our functions when everything went "right", the next object is also just for simulating the result when something goes "wrong".
+
+```kotlin:ank:silent
+object ErrorEffects : OurSideEffectsFunctions {
+
+    override fun validateLogin(cred: Credential): Either<SecurityError, Unit> {
+        return Either.left(SecurityError.RuntimeSecurityError("Invalid credentials"))
+    }
+
+    override fun userInfo(user: User): DeferredK<Profile> {
+        return DeferredK.failed(Throwable("Error retrieving profile"))
+    }
+}
+```
+
+If you want to run these examples you can run them inside a main body.
+
+```kotlin:ank:silent
+fun main(args: Array<String>) {
+
+}
+```
+
+If we need to extract the information for a List of Users we can create a function that composes with the function defined previously `userInfo`.
+
+```kotlin:ank
+fun profilesFor(users: List<User>): List<DeferredK<Profile>> = users.map { u -> ValidEffects.userInfo(u) }
+```
+
+Notice how we are returning a List of deferred computations. It would be nice for the ones using this function if we could aggregate the results and return the List of Profile under a single DeferredK, something like `DeferredK<List<Profile>>`
+
+To be able to do this transformation we have the `Traverse` type class.
+
+
+`Traverse` is defined with the following signature:
+
+interface Traverse<F> : Functor<F>, Foldable<F> {
+  fun <G, A, B> Kind<F, A>.traverse(AP: Applicative<G>, f: (A) -> Kind<G, B>): Kind<G, Kind<F, B>>
+}
+
+For our example, F would be `List` (the initial container), G would be the data type representing the side effect: `Option`, `Either` or `DeferredK`.
+
+So, if we have a `List<User>` (the ones we want to obtain their profiles) and a function `User -> DeferredK<Profile>`, with traverse we can transform and instead of obtaining a `List<DeferredK, Profile>` it can aggregate all the results to obtain a single `DeferredK<List,Profile>`.
+
+In this case, `traverse` can go over the collection, apply the function and aggregate the resulting values (with side effects) in a `List`.
+
+Basically, `F` is some sort of context which may contain a value. We are using List in the example, but there are `Traverse` implementations for `Option`, `Either` or `Validated`.
+
+Let's see another example further clarifying this.
+
+```kotlin:ank
+fun parseIntEither(s: String): Either<NumberFormatException, Int> =
+  Either.applicativeError<NumberFormatException>().catch(
```

Comment:
There's is a leap of faith here. The reader could not know about what's `ApplicativeError` yet and how it works. We should probably clarify that a bit before stepping in. I know many people already knows FP but many people comes to this docs to learn about all this things for the first time. We must always give them means to reach the information they require to avoid getting lost in a point like this one, therefore lower the learning curve as much as possible.

Reply:
There is no need to prefix `catch`. It is exported for Either as a extension function since 0.8.0. There is rarely the need to ever call `Whatever.typeClass()` any more because all the type class extension functions are exported for all data types that provide instances.

Reply:
I imported the instance (extension function) for catch, so the example should be simpler now.